From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Tue, 17 Oct 2017 15:16:24 +0200
Subject: [PATCH] commands: i2c: do not restrict i2c_write's data to be
 hex-formatted
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change the default base to 0 to be consistent with the "mw" command and
particularly allow decimally formatted data.

The price to pay is needing the usual prefix "0x" for each hex-formatted number.

Signed-off-by: Ulrich Ölmann <u.oelmann@pengutronix.de>
Acked-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
---
 commands/i2c.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/commands/i2c.c b/commands/i2c.c
index 573032ab1588..b74c53509f38 100644
--- a/commands/i2c.c
+++ b/commands/i2c.c
@@ -129,7 +129,7 @@ static int do_i2c_write(int argc, char *argv[])
 
 	buf = xmalloc(count);
 	for (i = 0; i < count; i++)
-		*(buf + i) = (char) simple_strtol(argv[optind+i], NULL, 16);
+		*(buf + i) = (char) simple_strtol(argv[optind+i], NULL, 0);
 
 	ret = i2c_write_reg(&client, reg | wide, buf, count);
 	if (ret != count) {
